pkg/gateway/operations/putobject.go

```diff
@@ -298,6 +298,11 @@ func handlePut(w http.ResponseWriter, req *http.Request, o *PathOperation) {
 	o.Incr("put_object", o.Principal, o.Repository.Name, o.Reference)
 	storageClass := StorageClassFromHeader(req.Header)
 	opts := block.PutOpts{StorageClass: storageClass}
+	err := o.checkIfAbsent(req)
```

Comment:
This intention is right but not the approach. 
In the current version the function `checkIfAbsent` implements `checkIfAbsent` by doing Get to the object and then Put. This is not gonna work because the Get->Put operation is not atomic. 
The idea of the feature is that AWS implements this and provide atomicity out of the box. 

Instead, our S3 Gateway should only forward the request to AWS and let them handle the precondition check. 
**Example** implementation: 
1. add additional field to `opts := block.PutOpts{StorageClass: storageClass, IfNoneMatch}` (one line above) that check if the header exist and sets it. 
2. let the ops propagate down to the blockstore adapter and then use it when calling [PutObject](https://github.com/treeverse/lakeFS/blob/master/pkg/block/s3/adapter.go#L268) as an option. We have a challenge here see below 👇 . 

## Challenge with PutObject call:
Our AWS SDK is outdated, so it does not contain the built-in option in the new version [IfNoneMatch](https://github.com/aws/aws-sdk-go-v2/blob/main/service/s3/api_op_PutObject.go#L418). 

### We have 2 options: 
a. Upgrade the AWS SDK (preferable) after verifying there no breaking changes (separate PR)
b. Use the request middleware options that the current AWS SDK version provides to propagate the header. 


# NOTE: Same should apply for Multipart Uploads